packages/react-query/src/useBaseQuery.ts

```diff
@@ -131,6 +132,21 @@ export function useBaseQuery<
     result,
   )
 
+  if (!isServer && willFetch(result, isRestoring) && !observer.hasListeners()) {
```

Comment:
should we hide this whole thing behind an `experimental_autoprefetch` option? We could make it a requirement to turn it on if yo want to work with the `experimental_promise`. 

The thing is I'm not sure if this fetching during render will not cause troubles for other cases that we can't forsee yet.

Reply:
alternatively, we could ship a new experimental hook that has this turned on ?